### PR TITLE
*: support plain insert on cluster index table

### PR DESCRIPTION
**Pull request by lysu**
### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary:

### What is changed and how it works?

support plain insert to a cluster index table and fully read it by table reader.

Proposal: [Clustered Index](https://github.com/pingcap/tidb/blob/master/docs/design/2020-05-08-cluster-index.md)

What's Changed:

What's Changed:

- upgrade tipb to add PrimaryCols info to tableReader and indexReader
- support simple insert/select on unistore and mocktikv

### Related changes

- n/a

### Check List 

Tests 

- Unit test

Side effects

- n/a

### Release note 

- No release note

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/17674)

### CODE REVIEW

File: store/mockstore/unistore/cophandler/cop_handler.go [hunk 1]
```diff
@@ -187,7 +189,14 @@ func (e *evalContext) newRowDecoder() (*rowcodec.ChunkDecoder, error) {
 		}
 		cols = append(cols, col)
 		if info.PkHandle {
-			handleColID = info.ColumnId
+			pkCols = append(pkCols, info.ColumnId)
+		}
+	}
+	if len(pkCols) == 0 {
+		if e.primaryCols != nil {
+			pkCols = append(pkCols, e.primaryCols...)
```
Comment (coocood) [inline store/mockstore/unistore/cophandler/cop_handler.go:197]: Can we assign instead of append?

File: table/tables/tables.go [hunk 2]
```diff
@@ -487,6 +487,28 @@ func (t *TableCommon) AddRecord(ctx sessionctx.Context, r []types.Datum, opts ..
...
+					pkIdx = idx
+					break
+				}
+			}
+			pkDts := make([]types.Datum, 0, len(pkIdx.Columns))
+			for _, idxCol := range pkIdx.Columns {
+				pkDts = append(pkDts, r[tblInfo.Columns[idxCol.Offset].Offset])
+			}
+			var handleBytes []byte
+			handleBytes, err = codec.EncodeKey(new(stmtctx.StatementContext), nil, pkDts...)
```
Comment (coocood) [inline table/tables/tables.go:503]: We need to get stmtctx from sessionctx.Context

File: executor/builder.go [hunk 3]
```diff
@@ -2992,6 +2992,22 @@ func newRowDecoder(ctx sessionctx.Context, schema *expression.Schema, tbl *model
...
 	}
+	if len(pkCols) == 0 {
+		if tbl.IsCommonHandle {
+			var pkIdx *model.IndexInfo
+			for _, idx := range tbl.Indices {
+				if idx.Primary {
+					pkIdx = idx
+					break
+				}
+			}
```
Comment (wshwsh12) [inline executor/builder.go:3003]: Could we use a function to do this, so that avoid coping the code 3 times in different place.

**Pull request merged**